packages/@aws-cdk/toolkit-lib/lib/toolkit/toolkit.ts

```diff
@@ -249,6 +252,145 @@ export class Toolkit extends CloudAssemblySourceBuilder implements AsyncDisposab
+          stack,
+          diffMethod.options.compareAgainstProcessedTemplate,
+        );
+        const currentTemplate = templateWithNestedStacks.deployedRootTemplate;
+        const nestedStacks = templateWithNestedStacks.nestedStacks;
+
+        const formatter = new DiffFormatter({
+          ioHelper,
+          oldTemplate: currentTemplate,
+          newTemplate: stack,
+        });
+
+        const migrator = new ResourceMigrator({ deployments, ioHelper });
+        const resourcesToImport = await migrator.tryGetResources(await deployments.resolveEnvironment(stack));
+        if (resourcesToImport) {
+          removeNonImportResources(stack);
+        }
+
+        let changeSet = undefined;
+
+        if (diffMethod.method === 'change-set') {
+          let stackExists = false;
+          try {
+            stackExists = await deployments.stackExists({
+              stack,
+              deployName: stack.stackName,
+              tryLookupRole: true,
+            });
+          } catch (e: any) {
+            await ioHelper.notify(IO.DEFAULT_TOOLKIT_DEBUG.msg(`Checking if the stack ${stack.stackName} exists before creating the changeset has failed, will base the diff on template differences.\n`));
+            await ioHelper.notify(IO.DEFAULT_TOOLKIT_DEBUG.msg(formatErrorMessage(e)));
+            stackExists = false;
+          }
+
+          if (stackExists) {
+            changeSet = await createDiffChangeSet(ioHelper, {
+              stack,
+              uuid: uuid.v4(),
+              deployments,
+              willExecute: false,
+              sdkProvider: await this.sdkProvider('diff'),
+              parameters: diffMethod.options.parameters ?? {},
+              resourcesToImport,
+            });
+          } else {
+            await ioHelper.notify(IO.DEFAULT_TOOLKIT_DEBUG.msg(`the stack '${stack.stackName}' has not been deployed to CloudFormation or describeStacks call failed, skipping changeset creation.`));
+          }
+        }
+
+        if (options.securityOnly) {
+          const securityDiff = formatter.formatSecurityDiff({
+            requireApproval: RequireApproval.BROADENING,
+            stackName: stack.displayName,
+            changeSet,
+          });
+          formattedSecurityDiff = securityDiff.formattedDiff ?? '';
+          diffs = securityDiff.formattedDiff ? diffs + 1 : diffs;
+        } else {
+          const diff = formatter.formatStackDiff({
+            strict,
+            context: contextLines,
+            stackName: stack.displayName,
+            changeSet,
+            isImport: !!resourcesToImport,
+            nestedStackTemplates: nestedStacks,
+          });
+          formattedStackDiff = diff.formattedDiff;
+          diffs = diff.numStacksWithChanges;
+        }
+      }
+    }
+    await ioHelper.notify(IO.CDK_TOOLKIT_I4401.msg(`✨ Number of stacks with differences: ${diffs}`, {
+      formattedSecurityDiff,
+      formattedStackDiff,
+    }));
+    await diffSpan.end();
```

Comment:
the first of these two message just is the the end. no need to doubling

Reply:
youre saying i dont need diffSpan.end()?

Reply:
You need `diffSpan.end()` but you are using that to send `CDK_TOOLKIT_I4401` with the payload

Reply:
or i need to end the span with the payload instead

Reply:
```ts
    await diffSpan.end(`✨ Number of stacks with differences: ${diffs}`, {
      formattedSecurityDiff,
      formattedStackDiff,
    });
```

Reply:
its the latter :)